packages/mui-material/src/Tabs/Tabs.js

```diff
@@ -614,29 +587,65 @@ const Tabs = React.forwardRef(function Tabs(inProps, ref) {
         resizeObserver.disconnect();
       }
     };
-  }, [updateIndicatorState, updateScrollButtonState]);
-
-  const handleTabsScroll = React.useMemo(
-    () =>
-      debounce(() => {
-        updateScrollButtonState();
-      }),
-    [updateScrollButtonState],
-  );
+  }, [updateIndicatorState]);
 
+  /**
+   * Toggle visibility of start and end scroll buttons
+   * Using IntersectionObserver on first and last Tabs.
+   */
   React.useEffect(() => {
+    let firstObserver;
+    let lastObserver;
+    const { 0: firstTab, length, [length - 1]: lastTab } = Array.from(tabListRef.current.children);
+    const threshold = 0.99;
+    const observerOptions = {
+      root: tabsRef.current,
+      threshold,
+    };
+
+    const handleScrollButtonStart = (entries) => {
+      let display = false;
+      entries.forEach(({ intersectionRatio }) => {
+        display = intersectionRatio < threshold;
```

Comment:
I am not very familiar with the IntersectionObserver API, but would it be better/simpler to check using [`isIntersecting` property](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry/isIntersecting)?

Reply:
Yes thank you :)